openfermioncirq/gates/fermionic_simulation.py

```diff
@@ -163,12 +322,55 @@ def __repr__(self):
                 ', '.join(cirq._compat.proper_repr(v) for v in self.weights),
                 exponent_str))
 
+    @classmethod
+    def wire_symbol(cls, use_unicode: bool):
+        return '↓↑' if use_unicode else 'a*a'
+
+    @classmethod
+    def from_interaction(cls,
+            operator: openfermion.InteractionOperator,
+            modes: Sequence[int]):
+        p, q = modes
+        tunneling_coeff = operator.one_body_tensor[p, q]
+        interaction_coeff = (
+                - operator.two_body_tensor[p, q, p, q]
+                + operator.two_body_tensor[q, p, p, q]
+                + operator.two_body_tensor[p, q, q, p]
+                - operator.two_body_tensor[q, p, q, p])
+        weights = (-tunneling_coeff, -interaction_coeff
+                ) # type: Tuple[complex, complex]
+        if any(weights):
+            return cls(weights)
+        return None
+
+    @property
+    def generator(self):
+        generator = np.zeros((4, 4), dtype=np.complex128)
+        # w0 |10><01| + h.c.
+        generator[2, 1] = self.weights[0]
+        generator[1, 2] = self.weights[0].conjugate()
+        # w1 |11><11|
+        generator[3, 3] = self.weights[1]
+        return generator
+
+    def to_interaction_operator(self,
+            operator: openfermion.InteractionOperator,
+            modes: Sequence[int]):
+        weights = tuple(w * self._exponent for w in self.weights)
+        operator.constant += self._exponent * self._global_shift
+        operator.one_body_tensor[modes] -= weights[0]
+        operator.one_body_tensor[modes[::-1]] -= weights[0].conjugate()
+        operator.two_body_tensor[tuple(modes) * 2] += weights[1]
+
+    def fswap(self, i: int = 0):
```

Comment:
Docstring.

Reply:
Added abstract method to `FermionicSimulationGate` and put docstring there.